src/notification/email/EMailNotificationTask.ts

```diff
@@ -141,87 +144,126 @@ export default class EMailNotificationTask implements NotificationTask {
     return this.prepareAndSendEmail('end-user-error-notification', data, user, tenant, severity);
   }
 
-  private async sendEmail(email, data, tenant: Tenant, user: User, severity: NotificationSeverity, retry = false): Promise<void> {
+  private async sendEmail(email: EmailNotificationMessage, data: any, tenant: Tenant, user: User, severity: NotificationSeverity, useBackup = false): Promise<void> {
+    // Email configuration sanity checks
+    if (!this.SMTPMainClientInstance) {
+      // No suitable main SMTP server configuration found to send the email
+      Logging.logError({
+        tenantID: tenant.id,
+        source: Utils.objectHasProperty(data, 'chargeBoxID') && data.chargeBoxID,
+        action: ServerAction.EMAIL_NOTIFICATION,
+        module: MODULE_NAME, method: 'sendEmail',
+        message: 'No suitable main SMTP server configuration found to send email',
+        actionOnUser: user
+      });
+      return;
+    }
+    // Toggle boolean attribute to switch SMTP server when necessary
+    if (!useBackup && this.backupInUse) {
```

Comment:
Remove 'if' statement by removing this.backupInUse and use the useBackup var to get the main or backup SMTP client (check next comment).

Reply:
That's debatable. The class in now a singleton so having the memory of SMTP instance in use make speculative branching works at all computer architecture level and the execution a lot faster. 
Let's keep that kind of optimisation for later to make the code more easily understandable.

Reply:
I would like it now as it'll simplify the code.

Reply:
I do not see what is missing since the previous review request. 
Could you please read the PR changelog and the latest commits patch?

Reply:
Remove this.backupInUse member variable and create 2 instances of the SMTP servers in the constructor.

Reply:
> Remove this.backupInUse member variable 

Already done since the previous review request. 
Could you please read the PR changelog in description and the latest commits patch?

> and create 2 instances of the SMTP servers in the constructor.

Quote of my previous response on that matter: 

'The implementation in the PR only create the SMTP client instance for the backup server in case of an error on the main SMTP server.
The code that do that small optimisation is easy to read and understand and I do not see why it should be replaced by a less optimised and readable code with syntactic sugar.'

Reply:
Already checked and read your comments!
backupInUse member variable is still there in GitHub:
![image](https://user-images.githubusercontent.com/25266945/106907687-d3702300-66fe-11eb-965c-98445da8597f.png)

I simply asked you to remove this handling of the SMTP servers in the first place and you created a singleton instead!
It's neither debatable nor to you to decide.
Remove it for the nth times.